### PR TITLE
Retrieve header bidding information from switch

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

Retrieve header bidding information from switches rather than from the config object.

Also involves the regeneration of fixtures to use a new config object that reflects the changes to header bidding properties in the config.

## Why?

To bring DCR in line with the corresponding [frontend PR](https://github.com/guardian/frontend/pull/23924) that exposes header bidding information via switches instead of via the config. This change has been made since the header bidding config properties were populated with information from switches anyway.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -672,7 +672,6 @@ interface ConfigType extends CommercialConfigType {
 	googletagUrl: string;
 	stage: string;
 	frontendAssetsFullURL: string;
-	hbImpl: { [key: string]: any } | string;
```
Comment (mxdvl) [inline index.d.ts:675]: This is the good stuff!

File: src/model/json-schema.json [hunk 2]
```diff
@@ -4141,4 +4129,4 @@
         }
     },
     "$schema": "http://json-schema.org/draft-07/schema#"
-}
```
Comment (mxdvl) [inline src/model/json-schema.json:4144]: Do we want to keep this as-is?

File: src/model/window-guardian.ts [hunk 3]
```diff
@@ -54,7 +54,10 @@ const makeWindowGuardianConfig = (
 			adUnit: config.adUnit,
 			showRelatedContent: true,
 			ajaxUrl: config.ajaxUrl,
-			hbImpl: config.hbImpl,
+			hbImpl: {
+				prebid: config.switches.prebidHeaderBidding,
+				a9: config.switches.a9HeaderBidding,
+			},
```
Comment (mxdvl) [inline src/model/window-guardian.ts:60]: Do we still need this? I think that it’s only needed in `dfpEnv` in the end.

**Review by mxdvl — CHANGES_REQUESTED** [review]

File: src/model/json-schema.json (hunk 2 above)
Comment (chrislomaxjones) [reply src/model/json-schema.json:4144]: I think we have to remove it from the schema otherwise DCR still expects a "hbImpl" property in the config

Comment (mxdvl) [reply src/model/json-schema.json:4144]: I was only referring to the return at the end of the file!

Comment (chrislomaxjones) [reply src/model/json-schema.json:4144]: Oh I see!

File: src/model/window-guardian.ts (hunk 3 above)
Comment (chrislomaxjones) [reply src/model/window-guardian.ts:60]: Good point I don't think its actually used anywhere in DCR?

**Review by mxdvl — APPROVED** [review]

**Pull request merged**